libraries/core_libs/network/src/tarcap/packets_handlers/latest/transaction_packet_handler.cpp

```diff
@@ -140,35 +151,125 @@ void TransactionPacketHandler::periodicSendTransactions(SharedTransactions &&tra
   }
 }
 
+std::vector<std::pair<std::shared_ptr<TaraxaPeer>, std::pair<SharedTransactions, std::vector<trx_hash_t>>>>
+TransactionPacketHandler::transactionsToSendToPeers(std::vector<SharedTransactions> &&transactions) {
+  // Main goal of the algorithm below is to send different transactions and hashes to different peers but still follow
+  // nonce ordering for single account and not send higher nonces without sending low nonces first
+  const auto accounts_size = transactions.size();
+  std::vector<std::pair<std::shared_ptr<TaraxaPeer>, std::pair<SharedTransactions, std::vector<trx_hash_t>>>>
+      peers_with_transactions_to_send;
+  auto peers = peers_state_->getAllPeers();
+
+  if (accounts_size) {
+    // trx_account_index keeps current account start index so that different peers will receive
+    // transactions from different accounts
+    uint32_t trx_account_index = 0;
+    for (const auto &peer : peers) {
+      if (!peer.second->syncing_) {
+        SharedTransactions peer_trxs;
+        std::vector<trx_hash_t> peer_trx_hashes;
+        bool trx_max_reached = false;
+        uint32_t trx_max_reached_index = 0;
+        auto trx_account_index_iterator = trx_account_index;
```

Comment:
It is an index, not an iterator. Shouldn't be this one a trx_account_index, but trx_account_index should have some different name?

Reply:
I made changes, please review again